AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,128 @@
+import http = require("http");
+
+import Logging = require("../Library/Logging");
+import TelemetryClient = require("../Library/TelemetryClient");
+
+class WebSnippet {
+
+    public static INSTANCE: WebSnippet;
+
+    private static _snippet: string;
+    private static _aiUrl: string;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+
+    constructor(client: TelemetryClient) {
+        if (!!WebSnippet.INSTANCE) {
+            throw new Error("Web snippet injection should be configured from the applicationInsights object");
+        }
+
+        WebSnippet.INSTANCE = this;
+        WebSnippet._aiUrl = "https://az416426.vo.msecnd.net/scripts/b/ai.2.min.js";
+        WebSnippet._snippet = '<script type="text/javascript">' +
```

Comment:
As this is a nodejs (server side) do we want to publish the snippet to the CDN so that this can fetch it and update dynamically? So that when we update the snippet we don't need to update this as well?

Reply:
Maybe we "publish" an autoinject_snippet.json or something, so that we could also change the default URL as well?

Reply:
Agree with doing something to make updating it easier. My take on this would be to add a `npm run snippet:update` script which goes and fetches  the snippet from `ApplicationInsights-JS` and writes it to some checked in file in this repo, `Snippet.ts`

Reply:
Added snippet retrieval logic